features/admin.email-providers.v1/api/email-providers.ts

```diff
@@ -37,9 +38,42 @@ const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance()
  *
  * @returns the email provider configurations of the tenant.
  */
-export const useEmailProviderConfig = <Data = EmailProviderConfigAPIResponseInterface[], Error = RequestErrorInterface>
+export const useEmailProviderConfig = <Data = EmailProviderConfigAPIResponseInterface[],Error = RequestErrorInterface>
     (): RequestResultInterface<Data, Error> => {
 
+    const enableOldUIForEmailProvider: boolean = useSelector(
+        (state: AppState) => state?.config?.ui?.enableOldUIForEmailProvider);
+
+    const requestConfig: RequestConfigInterface = {
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: enableOldUIForEmailProvider
+            ? store.getState().config.endpoints.emailProviderEndpoint
+            : store.getState().config.endpoints.emailProviderV2Endpoint
+    };
+
+    const { data, error, isValidating, mutate } = useRequest<Data, Error>(requestConfig);
+
+    return {
+        data,
+        error: error,
+        isLoading: !error && !data,
+        isValidating,
+        mutate: mutate
+    };
+};
+
+/**
+ * Get email provider configurations.
+ *
+ * @returns the email provider configurations of the tenant.
+ */
+export const usePasswordOfEmailProvider = <Data = EmailProviderConfigAPIResponseInterface[],
```

Comment:
doc comment and method name don't seem to be in sync. we may have update either one to match the other